### DISCUSSION TITLE
Add `reloadDocument` as a Link prop

### DISCUSSION THREAD

**Discussion by jasikpark** [TanStack/router#580] [Ideas]
A useful (though seemingly undocumented) feature of the ReactRouterv6 `<Link>`/`<NavLink>` components is the `reloadDocument` prop, which will do a traditional navigation for the link provided rather than an SPA navigation.

https://github.com/remix-run/react-router/blob/f85e0b33536645aa14c9d7cc6e3f9e09f63373da/packages/react-router-dom/index.tsx#L483

This is would be helpful for example when you want to implicitly upgrade an SPA's deployed version, without prompting the user to reload their browser or press a "upgrade app version" button.

**Comment by jasikpark** [comment]
ex:

```tsx
import { forwardRef } from 'react';
import { Link as ReactRouterLink, type LinkProps } from 'react-router-dom';
import { useGlobalState } from '@global/context/global-state/useGlobalState';

/**
 * Wraps the `Link` component from react-router: https://reactrouter.com/en/main/components/link
 *
 * The difference is that our version will turn into a real anchor tag to force a full page load if a new
 * version of the webapp is available.
 */
export const Link = forwardRef<HTMLAnchorElement, Omit<LinkProps, 'href'>>(
  ({ to, children, replace, reloadDocument, ...props }, ref) => {
    const { fullPageloadNeeded } = useGlobalState();

    return (
      <ReactRouterLink
        ref={ref}
        to={to}
        replace={replace}
        reloadDocument={fullPageloadNeeded || reloadDocument}
        {...props}
      >
        {children}
      </ReactRouterLink>
    );
  }
);
Link.displayName = 'Link';
```

but w/ tanstack router 😁

**Comment by schiller-manuel** [comment]
we have this now in https://github.com/TanStack/router/releases/tag/v1.88.0

**Reply by jasikpark** [reply]
woot! thx!
